### DISCUSSION TITLE
How to create/find/use TextureSwappableMaterial for Dynamic Re-texturing

### DISCUSSION THREAD

**Discussion by lyvelyte** [microsoft/AirSim#3990] [Support Q&A]
I'm working in Windows 10 with Unreal Engine 4.25, and the latest version of AirSim (both client and server) and python 3.8. 

I am trying to retexture objects using simSwapTextures. I've been able to change the parent class of my object to TextureShuffleActor and it has gained the member DynamicMaterial however the TextureSwappableMaterial is missing! Is there any extra steps needed to get this to show up or any reasons why it wouldn't automatically be present? 

Thanks for any help with this issue!

**Comment by jonyMarino** [comment]
Hi @lyvelyte! This is a question for @zimmy87.

**Comment by HarperReinhardt** [comment]
This is so incredibly late, but to anyone who finds this discussion, first double check that you have the latest version of airsim installed since the texture was only added in the most recent version. Next, press view options, then click show plugin content. This should show TextureSwappableMaterial in the list. 
![image](https://github.com/microsoft/AirSim/assets/127988667/17e2d1fa-e60a-4285-8691-0818b73a6cef)
![image](https://github.com/microsoft/AirSim/assets/127988667/99c1ef5f-0229-4d51-a020-ff9beaad6abe)
